transmat/mixins/buffer/fsBufferImpl.go

```diff
@@ -0,0 +1,85 @@
+package buffer
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+
+	. "github.com/warpfork/go-errcat"
+
+	api "go.polydawn.net/go-timeless-api"
+	"go.polydawn.net/go-timeless-api/rio"
+	"go.polydawn.net/rio/fs"
+	"go.polydawn.net/rio/fs/osfs"
+	"go.polydawn.net/rio/fsOp"
+)
+
+/*Buffer a ware to a locally seekable instance.
+ *Some forms of transmat need to know the length and seek through packed data.
+ *This buffer bridges the gap between that need at the `io.ReadCloser`
+ *supported by `warehouse.BlobstoreController`.
+ */
+type buffer struct {
+	base string
+	fs   fs.FS
+}
+
+func NewTemporaryBuffer() *buffer {
+	base := os.TempDir()
+	dir, err := ioutil.TempDir(base, "rio-")
```

Comment:
went with a cleaned up `ioutil.TempFile()`. Do you think it's worth making sure it gets placed in the right dir tree while on disk?

Reply:
I think the way you've written it now is fine.

While it's true that there's a bunch of other stuff that's a bit particular about the directory it's in... in most of the other cases, it's because those temp files are destined to be `mv`'d into place in the same (or sibling) directory: and so in those cases, it's important that the temp file be in nearly the right place because we surely wouldn't want that to be a cross-device move.  But that doesn't really apply here if we're certainly not keeping the file.

I think there might be a possibility for future work where we might decide to keep the packed zip file on hand (by comparison, looking at the git code can be very interesting -- it caches *several* different kinds of stuff at various layers of the git protocol).  But that's a whole 
'nother/bigger ball of wax.